app/models/requests/patron.rb

```diff
@@ -126,38 +131,71 @@ def load_patron(user:)
         end
       end
 
-      def current_patron(uid)
-        return false unless uid
-        begin
-          patron_record = Faraday.get "#{Requests.config[:bibdata_base]}/patron/#{uid}?ldap=true"
-        rescue Faraday::Error::ConnectionFailed
-          Rails.logger.info("Unable to connect to #{Requests.config[:bibdata_base]}")
-          return false
-        end
-        return false if patron_errors?(patron_record: patron_record, uid: uid)
-        JSON.parse(patron_record.body).with_indifferent_access
+      def bibdata_uri
+        Requests.config[:bibdata_base]
+      end
+
+      def build_patron_uri(uid:)
+        "#{bibdata_uri}/patron/#{uid}"
       end
 
-      def patron_errors?(patron_record:, uid:)
-        return false if patron_record.status == 200
-        if patron_record.status == 403
-          Rails.logger.info('403 Not Authorized to Connect to Patron Data Service at '\
-                      "#{Requests.config[:bibdata_base]}/patron/#{uid}")
-        elsif patron_record.status == 404
-          Rails.logger.info("404 Patron #{uid} cannot be found in the Patron Data Service.")
-        elsif patron_record.status == 500
-          Rails.logger.info('Error Patron Data Service.')
+      def api_request_patron(id:)
+        request_uri = build_patron_uri(uid: id)
+        response = Faraday.get("#{request_uri}?ldap=true")
+
+        case response.status
```

Comment:
@jrgriffiniii I find this odd that you are utilizing exceptions as part of the code logic within a single routine.  I could see if this routine did not handle the error, but as part of the same method it seems overly complex.  I like your use of a case statement over the elseif.  Maybe just move the actual code into the case?